libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2346,111 +2353,18 @@ string YulUtilFunctions::updateStorageValueFunction(
 
 			auto const& fromStructType = dynamic_cast<StructType const&>(_fromType);
 			auto const& toStructType = dynamic_cast<StructType const&>(_toType);
-			solAssert(fromStructType.structDefinition() == toStructType.structDefinition(), "");
 			solAssert(_offset.value_or(0) == 0, "");
 
 			Whiskers templ(R"(
 				function <functionName>(slot, <?dynamicOffset>offset, </dynamicOffset>value) {
 					<?dynamicOffset>if offset { <panic>() }</dynamicOffset>
-					<?fromStorage> if eq(slot, value) { leave } </fromStorage>
```

Comment:
Is this shortcut still there?

Reply:
Yes, it is inside of `copyStructToStorage` call